commands/command_prune.go

```diff
@@ -56,6 +56,19 @@ type PruneProgress struct {
 }
 type PruneProgressChan chan PruneProgress
 
+func maxConcurrency() int64 {
+	fds, err := tools.GetMaxFileDescriptors()
+	if err != nil {
+		return int64(runtime.NumCPU() * 2)
+	}
+	// Assume 64 FDs for each operation and leave 64 for other purposes.
+	maxThreads := (int64(fds) - 64) / 64
```

Comment:
Is it possible to set unlimited file descriptors? If yes, will this math behave properly?

Reply:
It is indeed possible to set the value to `RLIM_INFINITY`.  Assuming that's a positive and large number, that will work fine.  If it's a negative number, it will limit it to 1 thread, which, while not ideal, will also work.  I believe it's a positive and large number most places.